### ISSUE TITLE
adding neo4j-bolt config doesn't allow injection of Driver as stated in docs

### ISSUE THREAD

**Issue by jexp** [micronaut-projects/micronaut-core#261]
After adding `compile "io.micronaut.configuration:neo4j-bolt"` to build.gradle

and to application.yml

```
neo4j:
        uri: bolt://localhost
```

and running neo4j on the default bolt  port, I tried to inject the driver both as field injection and constructor injection. 

```
import org.neo4j.driver.v1.*;

@Singleton
public class CityRepository {
   private Driver driver;
   public @Inject CityRepository(Driver driver) { this.driver = driver; }
```

but in both cases it failed with

```
Message: No bean of type [org.neo4j.driver.v1.Driver] exists. If you are using Java or Kotlin make sure you have enabled annotation processing.
Path Taken: CityController.repo --> new CityRepository([Driver driver])
        at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:955)

```

**Comment by graemerocher** [comment]
Can you provide an example that reproduces this issue? Steps I took:

```
mn create-app meetup --features neo4j-bolt
cd meetup
```

Create class:

```java
import io.micronaut.context.annotation.Context;
import org.neo4j.driver.v1.Driver;

@Context
public class CityRepository {
    private Driver driver;
    CityRepository(Driver driver) {
         System.out.println("driver = " + driver);
        this.driver = driver;
    }
}
```

On application startup the driver is injected and he system out println executed

**Comment by jexp** [comment]
I did it manually, following the manual. 
I ran your example and saw the difference.

I had put the `neo4j` entry under `micronaut` not on the top level. Which seems to have caused this issue. Not sure if this should be clarified in the manual?

application.yml
```
micronaut:
...
       neo4j:
             uri: bolt://localhost
```

Can one add _features_ later with `mn`? Or only on creation?

Feel free to close if the docs don't need updating.
